tensorflow/lite/micro/kernels/reduce.cc

```diff
@@ -109,18 +130,76 @@ TfLiteStatus EvalMean(TfLiteContext* context, TfLiteNode* node) {
                                 GetTensorData<float>(output)));
       }
     } break;
+    case kTfLiteInt8: {
+      if (params->keep_dims) {
+        reference_integer_ops::Mean(
+            op_params, data->multiplier, data->shift, GetTensorShape(input),
+            GetTensorData<int8_t>(input), input->params.zero_point,
+            GetTensorShape(output), GetTensorData<int8_t>(output),
+            output->params.zero_point);
+      } else if (input->params.zero_point == output->params.zero_point &&
+                 input->params.scale == output->params.scale) {
+        TF_LITE_ENSURE(
+            context,
+            reference_ops::Mean(
+                GetTensorData<int8_t>(input), input->dims->data,
+                input->dims->size, GetTensorData<int8_t>(output),
+                output->dims->data, output->dims->size,
+                GetTensorData<int>(axis), num_axis, params->keep_dims,
+                temp_index, resolved_axis, GetTensorData<int>(output)));
```

Comment:
GetTensorData<int8_t>(output)

Reply:
The last argument is for the temporary sum that is then divided by the number of elements to make up the mean. Does this not need to be of type with more than 8 bits to handle the potential overflow of the sum? @hajuho

Reply:
I see. I missed it.
But, does the "output" tensor have enough capacity to store the array of int values? Temporary buffer is used for TFLite https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/reduce.cc#L428, but looks like GetTemporary is not available in micro https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/kernels/svdf.cc#L465.
@advaijain Do you know how to handle it?

Reply:
Sorry, I made a typo. @advaitjain Do you have any idea on this?

Reply:
Any update on this? @hajuho @advaitjain

Reply:
In latest commit I use context->RequestScratchBufferInArena to allocate this tensor